### DISCUSSION TITLE
$fetch bearer token with useAsyncData and pinia

### DISCUSSION THREAD

**Discussion by mehdikarimifa** [nuxt/nuxt#16288] [Questions] [answered]
Hello, I'm new to nuxt and was wondering if someone could help me out to approach this. I've an api end-point with ruby as backend which returns logged in user's data. I need to send user's token as header in $fetch request with useAsyncData so server renders app in authenticated mode for logged in users. Problem is the pinia getter which returns user's token is obviously null in server mode. How can I solve this issue?

**Comment by TheAlexLichter** [comment, marked as answer]
Hey! You can store the token in a cookie instead of `localStorage` so it is available on the server ☺️

**Reply by mehdikarimifa** [reply]
Thank you, this approach works as expected.
